src/main/java/frc/robot/Robot.java

```diff
@@ -36,62 +87,121 @@ public void robotInit() {
      */
     @Override
     public void robotPeriodic() {
-        //Logger.updateEntries();
-        Monologue.updateAll();
+        Robot.previousTimestamp = Robot.currentTimestamp;
+        Robot.currentTimestamp = Timer.getFPGATimestamp();
         CommandScheduler.getInstance().run();
-        robotContainer.periodic();
-        DriverUI.previousTimestmap = DriverUI.currentTimestamp;
-        DriverUI.currentTimestamp = Timer.getFPGATimestamp();
     }
 
     @Override
     public void disabledInit() {
+        Robot.gameMode = GameMode.DISABLED;
         robotContainer.onDisabled();
     }
-
+    
     @Override
-    public void disabledPeriodic() { }
+    public void disabledPeriodic() {
+        // Now while this may not necessarily be a constant...
+        // it needs to be updated.
+        DriverStation.refreshData();
+        Robot.alliance = DriverStation.getAlliance();
+    }
 
     @Override
-    public void autonomousInit() { 
+    public void disabledExit() {
+        // Shut off NetworkTables broadcasting for most logging calls
+        // if we are at competition
+        RobotContainer.gameModeStart = currentTimestamp;
+        // Monologue.setFileOnly(DriverStation.isFMSAttached());
+    }
+
+    @Override   
+    public void autonomousInit() {
+        DriveConstants.MAX_SPEED_METERS_PER_SECOND = AutoConstants.MAX_SPEED_METERS_PER_SECOND;
+        Robot.gameMode = GameMode.AUTONOMOUS;
         robotContainer.onEnabled();
+        // We only need to update alliance becuase
+        // sim GUI starts the bot in a "disconnected"
+        // state which won't update the alliance before
+        // we enable...
+        DriverStation.refreshData();
+        Robot.alliance = DriverStation.getAlliance();
+
         autonomousCommand = robotContainer.getAutonomousCommand();
-        
+
         if (autonomousCommand != null) {
             autonomousCommand.schedule();
         }
     }
 
     @Override
-    public void autonomousPeriodic() { }
-    
+    public void autonomousPeriodic() {
+    }
```

Comment:
Why override a method that already exists when this one doesn't have any implementation?

Reply:
Same question for teleopPeriodic() and teleopExit()

Reply:
You're totally right because it is not necessary, I just think it's nice to have the standard of every initialization and periodic method being there for transparency.

Reply:
So any call to that method will still use the original code from IterativeRobotBase?

Reply:
The original implementation of the mode-specific periodic methods like teleopPeriodic have no significant code in them, but they do have a silly print statement that I forgot about (another reason to override).